tests/test93.c

```diff
@@ -0,0 +1,123 @@
+#include <stdio.h>
+#include <setjmp.h>
+
+#define HASH_BLOOM 16
+
+#include "uthash.h"
+
+#undef uthash_malloc
+#undef uthash_fatal
+#define uthash_malloc(sz) alt_malloc(sz)
+#define uthash_fatal(s) alt_fatal(s)
+
+typedef struct example_user_t {
+    int id;
+    int cookie;
+    UT_hash_handle hh;
+} example_user_t;
+
+static int malloc_cnt = 0;
+static int malloc_failed;
+static int is_fatal;
+static jmp_buf j_buf;
+
+static void *alt_malloc(size_t sz)
+{
+    if (--malloc_cnt <= 0) {
+        // printf("fail malloc req for %ld\n", sz);
+        malloc_failed = 1;
+        return 0;
+    }
+    // printf("OK malloc req for %ld\n", sz);
+    malloc_failed = 0;
+    return malloc(sz);
+}
+
+static void alt_fatal(char const * s) {
+    // printf("alt_fatal: %s\n", s);
+    is_fatal = 1;
+    longjmp(j_buf, 1);
+}
+
+int main(int argc, char *argv[])
+{
+
+#define init(a) do { \
+    users = 0; \
+    user = (example_user_t*)malloc(sizeof(example_user_t)); \
+    user->id = id; \
+    is_fatal = 0; \
+    malloc_cnt = a; \
+    /* printf("adding to hash...\n"); */ \
+    if (!setjmp(j_buf)) { \
+        HASH_ADD_INT(users, id, user); \
+    } \
+} while(0)
+
+    // $TODO: this test doesn't work because
+    // uthash_fatal can't successfully return - it
+    // makes uthash write into *0. This either needs to
+    // be broken up into small test files, or use threads,
+    // a thread can hard exit. But there are no threads 
+    // in other tests, which means compilation changes, etc.
```

Comment:
The comment is no longer valid, I did remove it. By`write into *0` I meant that uthash would attempt writing data at address `0`, causing `SIGSEGV`